radixdlt-java/src/main/java/com/radixdlt/client/application/objects/Data.java

```diff
@@ -47,49 +45,50 @@ public DataBuilder publicReadable(boolean isPublicReadable) {
 
 		public Data build() {
 			final byte[] bytes;
-			final List<EncryptedPrivateKey> protectors;
+			final Encryptor encryptor;
 
 			if (isPublicReadable) {
-				protectors = Collections.emptyList();
+				encryptor = null;
 				bytes = this.bytes;
 				metaData.put("encrypted", false);
 			} else {
-				if (readers.isEmpty()) {
+				if (encryptorBuilder.getNumReaders() == 0) {
```

Comment:
Would this validation be better in the `build()` method of `EncryptorBuilder`?

Reply:
I don't think so. An encryptor should be able to be initialized with no readers but then updated to include more readers later on. I can see the same logic with encrypted data so this check is really just a sanity check for now.

Reply:
But `Encryptor` doesn't provide the functionality to add readers after the fact?

Reply:
OK, but I'm not sure what use an Encryptor with 0 readers is?

Given the way things are now, it doesn't seem possible to add anything to Encryptor#protectors after construction.